source/val/validate_memory.cpp

```diff
@@ -1410,7 +1410,60 @@ spv_result_t ValidatePtrAccessChain(ValidationState_t& _,
              << "VariablePointers or VariablePointersStorageBuffer";
     }
   }
-  return ValidateAccessChain(_, inst);
+
+  // Need to call first, will make sure Base is a valid ID
+  if (auto error = ValidateAccessChain(_, inst)) return error;
+
+  const auto base_id = inst->GetOperandAs<uint32_t>(2);
+  const auto base = _.FindDef(base_id);
+  const auto base_type = _.FindDef(base->type_id());
+  const auto base_type_storage_class = base_type->word(2);
+
+  if (_.HasCapability(SpvCapabilityShader) &&
+      (base_type_storage_class == SpvStorageClassUniform ||
+       base_type_storage_class == SpvStorageClassStorageBuffer ||
+       base_type_storage_class == SpvStorageClassPhysicalStorageBuffer ||
+       base_type_storage_class == SpvStorageClassPushConstant) &&
+      !_.HasDecoration(base_type->id(), SpvDecorationArrayStride)) {
```

Comment:
This isn't wrong, but it would be good to include explicit workgroup layout here too if possible.

```suggestion
  if (_.HasCapability(SpvCapabilityShader) &&
      (base_type_storage_class == SpvStorageClassUniform ||
       base_type_storage_class == SpvStorageClassStorageBuffer ||
       base_type_storage_class == SpvStorageClassPhysicalStorageBuffer ||
       base_type_storage_class == SpvStorageClassPushConstant ||
       (_.HasCapability(WorkgroupMemoryExplicitLayoutKHR) &&
        base_type_storage_class == SpvStorageClassWorkgroup)) &&
      !_.HasDecoration(base_type->id(), SpvDecorationArrayStride)) {
```